yarn-project/pxe/src/pxe_oracle_interface/pxe_oracle_interface.test.ts

```diff
@@ -563,6 +565,109 @@ describe('PXEOracleInterface', () => {
     });
   });
 
+  describe('getLogByTag', () => {
+    const tag = Fr.random();
+    let scopedLog: TxScopedL2Log;
+    let txEffect: TxEffect;
+
+    beforeEach(async () => {
+      scopedLog = await TxScopedL2Log.random();
+      txEffect = await TxEffect.random();
+      aztecNode.getLogsByTags.mockReset();
+      aztecNode.getTxEffect.mockReset();
+    });
+
+    it('returns null if no logs found for tag', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[]]);
+
+      const result = await pxeOracleInterface.getLogByTag(tag);
+      expect(result).toBeNull();
+    });
+
+    it('returns log data when single log found', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog]]);
+      aztecNode.getTxEffect.mockResolvedValue(randomInBlock(txEffect));
+
+      const result = await pxeOracleInterface.getLogByTag(tag);
+
+      // The implementation returns hex strings, not Buffer objects
+      expect(result?.logContent).toEqual(scopedLog.log.toFields());
+      expect(result?.uniqueNoteHashesInTx).toEqual(txEffect.noteHashes);
+      expect(result?.txHash).toEqual(scopedLog.txHash);
+      expect(result?.firstNullifierInTx).toEqual(txEffect.nullifiers[0]);
+
+      expect(aztecNode.getLogsByTags).toHaveBeenCalledWith([tag]);
+      expect(aztecNode.getTxEffect).toHaveBeenCalledWith(scopedLog.txHash);
+    });
+
+    it('throws if multiple logs found for tag', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog, scopedLog]]);
+
+      await expect(pxeOracleInterface.getLogByTag(tag)).rejects.toThrow(/Got 2 logs for tag/);
+    });
+
+    it('throws if tx effect not found', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog]]);
+      aztecNode.getTxEffect.mockResolvedValue(undefined);
+
+      await expect(pxeOracleInterface.getLogByTag(tag)).rejects.toThrow(/failed to retrieve tx effects/);
+    });
+
+    it('trims zero padding from log fields', async () => {
+      const logContent = [Fr.random(), Fr.random()];
+      const logContentWithPadding = [
+        ...logContent,
+        ...Array(PRIVATE_LOG_SIZE_IN_FIELDS - logContent.length).fill(Fr.ZERO),
+      ] as Tuple<Fr, typeof PRIVATE_LOG_SIZE_IN_FIELDS>;
+
+      const scopedLogWithPadding = new TxScopedL2Log(
+        TxHash.random(),
+        randomInt(100),
+        randomInt(100),
+        randomInt(100),
+        new PrivateLog(logContentWithPadding),
+      );
+
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLogWithPadding]]);
+      aztecNode.getTxEffect.mockResolvedValue(randomInBlock(txEffect));
+
+      const result = await pxeOracleInterface.getLogByTag(tag);
+
+      expect(result?.logContent).toEqual(logContent);
+    });
+  });
+
+  describe('removeNullifiedNotes', () => {
```

Comment:
Some properties are not being tested:

- that we're querying nullifiers at the correct (synched) block. spoilers: we're actually not, I'm fairly certain this is a bug since we'd be inserting 'future' state into PXE.
- that we're searching for notes from all accounts

Reply:
Good points. Addressed the synced block issue in a PR down the stack and added the tests for it in this PR.

Also added the all-accounts check in https://github.com/AztecProtocol/aztec-packages/pull/13323/commits/f80e247d20d338ff9b0d1cb351c9dcdb02829110

The spy is good to use in that test, agree?